### PR TITLE
Fix bug in node controller：when MarkPodsNotReady，we should check  if currentReadyCondition.Status equal to true

### PR DESCRIPTION
**Pull request by mysunshine92**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
when we do MarkPodsNotReady，we should check currentReadyCondition，not observedReadyCondition.

#### Which issue(s) this PR fixes:

Fixes #
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by mysunshine92** [comment]
cc @SergeyKanzhelev

File: pkg/controller/nodelifecycle/node_lifecycle_controller.go [hunk 1]
```diff
@@ -840,7 +840,7 @@ func (nc *Controller) monitorNodeHealth(ctx context.Context) error {
 				// Report node event only once when status changed.
 				controllerutil.RecordNodeStatusChange(nc.recorder, node, "NodeNotReady")
 				fallthrough
-			case needsRetry && observedReadyCondition.Status != v1.ConditionTrue:
+			case needsRetry && currentReadyCondition.Status != v1.ConditionTrue:
```
Comment (249043822) [inline pkg/controller/nodelifecycle/node_lifecycle_controller.go:843]: Suppose there are the following cases:
1. in current iteration, currentReadyCondition.Status != v1.ConditionTrue and observedReadyCondition.Status == v1.ConditionTrue, that means Ready -> NotReady, then fallthrough to MarkPodNotReady, if mark failed, then store to nodeRetry
2. in the next iteration,  if the node becomes ready, that means observedReadyCondition and currentReadyCondition are both ready, if the node is still not ready, then currentReadyCondition could not become ready(see `tryUpdateNodeHealth`, Ready ConditionTrue can only set by kubelet itself)
so we need not do this change

**Comment by SergeyKanzhelev** [comment]
/close

Thank you for your contribution. In current form PR lack any information what the bug is and why this change is needed. Also no tests were added.

Can you please add more details and re-open PR. Or link/file a bug with the observed issue. Thank you again!

**Pull request closed without merging**